### PR TITLE
doc: update documentation for release 3.6.0

### PR DESCRIPTION
**Pull request by humblec**
This commit add upgrade documentation for release 3.6.0
and also update support matrix for v3.6.0.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

File: docs/ceph-csi-upgrade.md [hunk 1]
```diff
@@ -73,6 +74,11 @@ to upgrade from cephcsi v3.3 to v3.4
 
 ## Upgrading from v3.4 to v3.5
 
+Refer [upgrade-from-v3.4-v3.5](https://github.com/ceph/ceph-csi/blob/v3.5.0/docs/ceph-csi-upgrade.md)
```
Comment (Madhu-1) [inline docs/ceph-csi-upgrade.md]:
```suggestion
Refer [upgrade-from-v3.4-v3.5](https://github.com/ceph/ceph-csi/blob/v3.5.1/docs/ceph-csi-upgrade.md)
```

**Review by Madhu-1 — COMMENTED** [review]
small nit.

File: docs/ceph-csi-upgrade.md (hunk 1 above)
Comment (humblec) [reply docs/ceph-csi-upgrade.md]: :+1:  thanks

Comment (humblec) [reply docs/ceph-csi-upgrade.md]: Addressed it .. @Madhu-1

**Review by Madhu-1 — APPROVED** [review]

**Review by yati1998 — APPROVED** [review]

**Pull request merged**